index.js

```diff
@@ -178,13 +178,23 @@ function setupContext (appName, opts, cb) {
     }
   }, opts))
 
-  console.log(ssbConfig)
 
   ssbConfig.keys = ssbKeys.loadOrCreateSync(Path.join(ssbConfig.path, 'secret'))
 
-  // fix offline on windows by specifying 127.0.0.1 instead of localhost (default)
-  var id = ssbConfig.keys.id
-  ssbConfig.remote = `net:127.0.0.1:${ssbConfig.port}~shs:${id.slice(1).replace('.ed25519', '')}`
+  const keys = ssbConfig.keys
+  const pubkey = keys.id.slice(1).replace(`.${keys.curve}`, '')
+
+
+  if (process.platform === 'win32') {
+    // fix offline on windows by specifying 127.0.0.1 instead of localhost (default)
+    ssbConfig.remote = `net:127.0.0.1:${config.port}~shs:${pubkey}`
+  } else {
+    const socketPath = Path.join(ssbConfig.path, 'socket')
+    ssbConfig.connections.incoming.unix = [{ 'scope': 'local', 'transform': 'noauth' }]
+    ssbConfig.remote = `unix:${socketPath}:~noauth:${pubkey}`
+  }
+
+  console.log(ssbConfig)
```

Comment:
Shamelessly stolen from https://github.com/ssbc/patchbay/pull/264

Reply:
@mixmix I see that we're doing it [differently](https://github.com/ssbc/patchbay/blob/master/config.js) in Patchbay, so this section may require a revamp. I've only tested this on Linux where the unix socket works great.

Reply:
Note to future travelers: this exposes your private key to the terminal output. It's harmless by itself, but if you copy and paste your terminal output from this commit to GitHub then you may expose your private key.

This bug has been fixed in `master`, but it's still in the Git history (obviously).